sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2714,6 +2714,14 @@ object SQLConf {
     .stringConf
     .createWithDefault("avro,csv,json,kafka,orc,parquet,text")
 
+  val ALLOW_EMPTY_SCHEMAS_FOR_WRITES =
+    buildConf("spark.sql.sources.file.allowEmptySchemaWrite")
```

Comment:
Other configurations use `spark.sql.sources.files` instead of `file` as a singular. Can you update it to be consistent?

On a related note, why does this include `sources`? At a glance `spark.sql.files.allowEmptySchemaWrite` seems to fully define the configuration. If this is okay, this should live with the rest of the `spark.sql.files.` configs. Let me know if there is a defined convention that I'm not aware of.